metagraph/src/cli/load/load_graph.cpp

```diff
@@ -71,5 +71,15 @@ std::shared_ptr<DeBruijnGraph> load_critical_dbg(const std::string &filename) {
     exit(1);
 }
 
+CanonicalDBG primary_to_canonical(std::shared_ptr<DeBruijnGraph> graph, size_t cache_size) {
+    if (graph->get_mode() != DeBruijnGraph::PRIMARY) {
+        logger->error("Only primary mode graphs can be wrapped into canonical mode.");
+        exit(1);
+    }
```

Comment:
this should be an assertion, which could actually happen in the constructor of CanonicalDBG itself. After this, the function becomes 2 lines. Probably not needed...